packages/core/src/indexing-store/utils/cursor.ts

```diff
@@ -0,0 +1,71 @@
+import { deserialize, serialize } from "@/utils/serialize.js";
+import type { ExpressionBuilder } from "kysely";
+import type { Row } from "../store.js";
+import type { OrderByConditions } from "./sort.js";
+
+export function encodeCursor(
+  record: Row,
+  orderByConditions: OrderByConditions,
+) {
+  const cursorValues = orderByConditions.map(([columnName]) => [
+    columnName,
+    record[columnName],
+  ]);
+
+  return Buffer.from(serialize(cursorValues)).toString("base64");
```

Comment:
What is the purpose of encoding as a buffer instead of just using the string returned from `serialize`?

Reply:
Cursors should not be human/client-readable. You don't want to encourage clients of the API to use the cursor value for anything other than passing it back to the server. So, it's best to encode it as an opaque string.